src/hooks/useDatasetManager.js

```diff
@@ -1,172 +1,216 @@
-    const response = await api.dataset.get(tempDatasetId) // TEMPORRAY
-    setDataset({
-      ...response
-    })
-    setLoading(false)
-  }
-
-  // fetches the dataset with the query 'details'
-  // details: https://github.com/AlexsLemonade/refinebio-frontend/pull/485
-  const getDatasetDetails = async () => {
-    if (!datasetId) {
-      return
+    const headers = token
+      ? {
+          'APT-KEY': token
+        }
+      : {}
+    const response = await api.dataset.get(id || datasetId, headers)
+
+    const formattedResponse = {
+      ...response,
+      experiments: formatExperiments(response.experiments)
     }
 
-    const params = {
-      details: true
+    if (!id) {
+      setDataset(formattedResponse)
     }
 
-    setLoading(true)
-    const response = await api.dataset.get(tempDatasetId, params) // TEMPORRAY
-    setDataset({
-      ...response,
-      experiments: formatExperiments(response.experiments)
-    })
     setLoading(false)
-  }
 
-  const emptyDataset = async () => {
-    const data = {}
+    return formattedResponse
+  }
 
+  const clearDataset = async (id = '') => {
     setLoading(true)
-    const response = await api.dataset.update(tempDatasetId, { data }) // TEMPORARY
+    const params = { data: {} }
+    const response = await api.dataset.update(id || datasetId, params)
+
     setDataset(response)
     setLoading(false)
   }
 
-  const updateDataset = async (datasetSlice) => {
-    if (!datasetId) {
-      createDataset()
-    }
-
-    const data = dataset ? { ...dataset.data } : {}
+  const updateDataset = async (id, params) => {
+    const isCurrentDatasetId = id === datasetId
+    const response = await api.dataset.update(id, params)
 
-    for (const accessionCode of Object.keys(datasetSlice)) {
-      if (datasetSlice[accessionCode].all) {
-        // the special key 'ALL' to add all samples from an experiment
-        // ALL: https://github.com/AlexsLemonade/refinebio-frontend/issues/496#issuecomment-456543865
-        data[accessionCode] = ['ALL']
-      } else {
-        data[accessionCode] = unionizeArrays(
-          data[accessionCode] || [],
-          datasetSlice[accessionCode]
-        )
-      }
+    if (isCurrentDatasetId) {
+      setDataset({
+        ...response,
+        experiments: formatExperiments(response.experiments)
+      })
     }
 
-    setLoading(true)
-    const response = await api.dataset.update(tempDatasetId, { data }) // TEMPORRAY
-    setDataset(response)
-    setLoading(false)
+    return response
   }
 
   /* Experiment */
-  const removeExperiment = async (experimentAccessionCode, details = false) => {
-    const data = {}
+  const getTotalExperiments = (data) =>
+    isEmptyObject(data) ? 0 : Object.keys(data).length
+
+  // formats the sample and experiment arrays from the API response
+  // to objects with experiment accession codes as their keys for UI
+  const formatExperiments = (experiments = []) => {
+    if (!experiments.length) return []
```

Comment:
It seems like this function should always return an object? I can't tell if an empty object or the argument `experiments` should be return if it is empty.

Also, I am confused about this function's purpose. It seems like the API's default response is in this format. If this method was called with an empty object it would return an empty array. Below is an example data attribute from a dataset detail view.

```json
  "data": {
    "GSE62528": [
      "GSM1528412",
      "GSM1528409",
      "GSM1528414",
      "GSM1528410",
      "GSM1528413",
      "GSM1528411"
    ]
  },
```

Reply:
Ah, this method is used for the `experiments` array in the API response and not the `data` object (when fetching the dataset with the `details` string query returns the additional properties `experiments` and `organism_samples`).  And an edge case was added so that it returns an empty array if the "experiments" array is empty to prevent an error. 

We use this method to format `experiments` for UI 🐨

e.g. ) 
Before the formatting:
```javascript
[ 
   { …, accession_code: "GSE116436", ... }, 
   { …, accession_code: "SRP066613", ... } 
] 
```

After the formatting:
```javascript
{ 
   GSE116436: {…}, 
   SRP066613: {…} 
} 
```

Reply:
Previously we decided to pass the `details` query to fetch a dataset details view (with `experiments` and `organism_samples`) on every API call and removed the `details` parameter option and `getDatasetDetails` method (https://github.com/AlexsLemonade/refinebio-web/issues/178#issuecomment-1719684364),  but this method was used like so (maybe it was clear as to why we need this):

```js

  // fetches the dataset with the query 'details'
  // details: https://github.com/AlexsLemonade/refinebio-frontend/pull/485
  const getDatasetDetails = async (id = '') => {
    const headers = token
      ? {
          'APT-KEY': token
        }
      : {}
    const response = await api.dataset.get(id, { details: true }, headers)

    const formattedResponse = {
      ...response,
      experiments: formatExperiments(response.experiments) // we reformat the experiments array for UI
    }

    return formattedResponse
  }

```

And due to this update, now this method is used by other methods that make API call to fetch a dataset. For instance, please see `getDataset` method:

https://github.com/AlexsLemonade/refinebio-web/blob/ca24c286f3252a807512e1f7d35ed2a58a9d545f/src/hooks/useDatasetManager.js#L42-L47

Reply:
Ok, this seems fine, please have the function return an empty object if it is empty.

Reply:
```suggestion
    if (!experiments.length) return {}
```